compiler/IREmitter/Payload/codegen-payload.c

```diff
@@ -2319,33 +2324,45 @@ void sorbet_throwReturn(VALUE retval) {
 // because this function will be inlined, meaning that `tag->buf` will reflect the
 // state of this function's caller and `tag->buf` will be valid for the lifetime of
 // this function's caller.
+//
+// The `volatile` here is a bit gross.  If you look at `rb_ec_tag_state` and its
+// accompanying logic, referenced below, you'll notice that it contains some
+// gross hacks to force clang to stick `ec` in memory before accessing it and that
```

Comment:
Is `VAR_FROM_MEMORY` the hack you're referring to here? (Might give a pointer to it if so.)